### PR TITLE
Document how to use roaring bitmaps

### LINKED ISSUE #2408: `indexSpec` tuning for ingestion is not documented
I couldn't find the documentation for `indexSpec` tuning (roaring / concise, lz4/lzf/uncompressed) as per https://groups.google.com/d/msg/druid-user/WJnyrPVwmso/bD4Lh1-6BgAJ

This should be documented.

### PR DESCRIPTION
**Pull request by fjy**
This fixes #2408.
While not all indexSpec properties are explained, it does explain how roaring bitmaps can be turned on.

### CODE REVIEW

**Comment by fjy** [comment]
I +1ed the original PR in #2468

File: docs/content/ingestion/stream-pull.md [hunk 1]
```diff
@@ -99,6 +99,9 @@ The property `druid.realtime.specFile` has the path of a file (absolute or relat
       "basePersistDirectory": "\/tmp\/realtime\/basePersist",
       "rejectionPolicy": {
         "type": "serverTime"
+      },
+      "indexSpec": {
+         "bitmap":"roaring"
```
Comment (gianm) [inline docs/content/ingestion/stream-pull.md]: please don't include non-default tunings in examples, this just causes confusion.

Comment (gianm) [inline docs/content/ingestion/stream-pull.md]: (I learned this the hard way in the past)

File: docs/content/ingestion/stream-pull.md [hunk 2]
```diff
@@ -166,6 +172,13 @@ The following policies are available:
 * `messageTime` &ndash; Can be used for non-"current time" as long as that data is relatively in sequence. Events are rejected if they are less than `windowPeriod` from the event with the latest timestamp. Hand off only occurs if an event is seen after the segmentGranularity and `windowPeriod` (hand off will not periodically occur unless you have a constant stream of data).
 * `none` &ndash; All events are accepted. Never hands off data unless shutdown() is called on the configured firehose.
 
+### Index Spec
+
+|Field|Type|Description|Required|
+|-----|----|-----------|--------|
+|bitmap|string|type of bitmap to use (e.g. roaring or concise), null to use the default.Defaults to the bitmap type specified by the (deprecated) "druid.processing.bitmap.type" setting| No|
```
Comment (gianm) [inline docs/content/ingestion/stream-pull.md]: missing space after period.

Also why is this doc different from the other one? Aren't they interpreted the same way and probably should be documented the same way?

File: docs/content/ingestion/stream-pull.md [hunk 3]
```diff
@@ -166,6 +172,13 @@ The following policies are available:
...
+|dimensionCompression|string|compression format for dimension columns. The default, null, means no compression|No|
```
Comment (drcrallen) [inline docs/content/ingestion/stream-pull.md]: Null means `io.druid.segment.data.CompressedObjectStrategy#DEFAULT_COMPRESSION_STRATEGY` which is LZ4

File: docs/content/ingestion/stream-pull.md (hunk 1 above)
Comment (gianm) [inline docs/content/ingestion/stream-pull.md]: it'd actually be good to change `maxRowsInMemory` to the default too.

File: docs/content/ingestion/stream-pull.md (hunk 2 above)
Comment (drcrallen) [inline docs/content/ingestion/stream-pull.md]: defaults to concise in `io.druid.segment.IndexSpec#IndexSpec(io.druid.segment.data.BitmapSerdeFactory, java.lang.String, java.lang.String)`

File: docs/content/ingestion/stream-pull.md [hunk 4]
```diff
@@ -155,6 +158,9 @@ The tuningConfig is optional and default parameters will be used if no tuningCon
...
 |reportParseExceptions|Boolean|If true, exceptions encountered during parsing will be thrown and will halt ingestion. If false, unparseable rows and fields will be skipped. If an entire row is skipped, the "unparseable" counter will be incremented. If some fields in a row were parseable and some were not, the parseable fields will be indexed and the "unparseable" counter will not be incremented.|false|
 |handoffConditionTimeout|long|Milliseconds to wait for segment handoff. It must be >= 0 and 0 means wait forerver.|0|
+|indexSpec|Object|Tune how data is indexed. See below for more information.|no|
+|reportParseExceptions|Boolean|If true, exceptions encountered during parsing will be thrown and will halt ingestion. If false, unparseable rows and fields will be skipped. If an entire row is skipped, the "unparseable" counter will be incremented. If some fields in a row were parseable and some were not, the parseable fields will be indexed and the "unparseable" counter will not be incremented.|false|
```
Comment (gianm) [inline docs/content/ingestion/stream-pull.md]: `reportParseExceptions` and `handoffConditionTimeout` are already documented earlier in this table.

File: docs/content/ingestion/stream-pull.md [hunk 5]
```diff
@@ -166,6 +167,13 @@ The following policies are available:
 * `messageTime` &ndash; Can be used for non-"current time" as long as that data is relatively in sequence. Events are rejected if they are less than `windowPeriod` from the event with the latest timestamp. Hand off only occurs if an event is seen after the segmentGranularity and `windowPeriod` (hand off will not periodically occur unless you have a constant stream of data).
 * `none` &ndash; All events are accepted. Never hands off data unless shutdown() is called on the configured firehose.
 
+### Index Spec
+
+|Field|Type|Description|Required|
+|-----|----|-----------|--------|
+|bitmap|String|The type of bitmap index to create. Choose from `roaring` or `concise`, or null to use the default (`concise`).|No|
+|dimensionCompression|String|Compression format for dimension columns. Choose from `LZ4`, `LZF`, or `uncompressed`. The default is `LZ4`.|No|
+|metricCompression|String|Compression format for dimension columns. Choose from `LZ4`, `LZF`, or `uncompressed`. The default is `LZ4`.|No|
```
Comment (gianm) [inline docs/content/ingestion/stream-pull.md]: metricCompression probably doesn't control compression for dimension columns.

**Comment by gianm** [comment]
👍

**Pull request merged**